containers/src/Kokkos_DualView.hpp

```diff
@@ -483,7 +493,7 @@ template< class , class , class , class > friend class DualView ;
   void sync ( const typename Impl::enable_if<
       ( ! std::is_same< typename traits::data_type , typename traits::non_const_data_type>::value ) ||
       ( std::is_same< Device , int>::value)
-      , int >::type& = 0 )
+      , int >::type& = 0 ) const
```

Comment:
If we mark `sync` const, that implies we can't later change DualView to do lazy allocation on one side.